rove/src/data_switch.rs

```diff
@@ -105,14 +120,15 @@ impl<'ds> DataSwitch<'ds> {
     // TODO: handle backing sources
     pub async fn get_spatial_data(
         &self,
-        source_id: &str,
+        polygon: Vec<GeoPoint>,
+        extra_spec: &str,
         timestamp: Timestamp,
     ) -> Result<SpatialCache, Error> {
         let data_source = self
             .sources
-            .get(source_id)
-            .ok_or_else(|| Error::InvalidDataSource(source_id.to_string()))?;
+            .get(extra_spec)
+            .ok_or_else(|| Error::InvalidDataSource(extra_spec.to_string()))?;
```

Comment:
Uhhhh we still need source_id here?! This bit of code is figuring out which data source we're trying to get data from, and routing to its connector. Sources is a hashmap of strings to DataSource implementors, so in your case "frost" to Frost. So you should be passing in the source_id as a key, which is "frost", not the extra_spec, which will be "air_temperature" or some such

Reply:
Ah, source_id is frost --- ok somehow I thought it was something that was only for series, to do with a station number and element or some such thing...

Reply:
Yeah series has something more complicated, series_id, which is "source_id:timeseries_id". We split on the colon and use source_id to index the hashmap, while timeseries_id gets passed to the connector.

Perhaps we ought to do something similar in spatial and combine data_source with extra_spec?

Reply:
Depends on if you prefer more arguments, or the complexity of parsing the string?

Reply:
Yeah, either way we should be consistent between series and spatial. Let’s discuss this in person?

Reply:
Now we have data_source, but it still has to contain the element... so it could really be called the same thing as in the get_series_data?

Reply:
So, potentially change this data_source to spatial_id, but since the polygon is seperate its not truly the whole id...

Reply:
I think that's fine though 🤷‍♀️ the polygon is optional anyway, you're still defining a spatial slice. Unless you have any better ideas?